wsmaster/che-core-api-factory-bitbucket-server/src/main/java/org/eclipse/che/api/factory/server/bitbucket/server/NoopBitbucketServerApiClient.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.factory.server.bitbucket.server;
+
+import java.util.List;
+import java.util.Set;
+import org.eclipse.che.api.factory.server.scm.exception.ScmBadRequestException;
+import org.eclipse.che.api.factory.server.scm.exception.ScmCommunicationException;
+import org.eclipse.che.api.factory.server.scm.exception.ScmItemNotFoundException;
+import org.eclipse.che.api.factory.server.scm.exception.ScmUnauthorizedException;
+import org.eclipse.che.commons.subject.Subject;
+
+/**
+ * Implementation of @{@link BitbucketServerApiClient} that is going to be deployed in container in
+ * case if no integration with Bitbucket server is needed.
+ */
+public class NoopBitbucketServerApiClient implements BitbucketServerApiClient {
+  @Override
+  public boolean isConnected(String bitbucketServerUrl) {
+    return false;
+  }
+
+  @Override
+  public BitbucketUser getUser(Subject cheUser)
+      throws ScmUnauthorizedException, ScmCommunicationException {
+    throw new RuntimeException("Invalid usage of BitbucketServerApi");
```

Comment:
Same as with `NoopOAuthAuthenticator`, IMHO this should throw a different kind of exception with a more detailed description why we might have gotten here.